lib/cell.rb

```diff
@@ -4,22 +4,43 @@ class Cell
   def initialize(coordinate)
     @coordinate = coordinate
     @ship = nil
+    @fired_upon = false
   end
 
   def empty?
-    @ship.nil?
+    if @ship == nil
+      true
+    else
+      false
+    end
   end
 
   def place_ship(ship)
     @ship = ship
   end
 
   def fired_upon?
-    @ship.health < @ship.length
+    @fired_upon
   end
 
   def fire_upon
-    @ship.hit
+    @fired_upon = true
```

Comment:
Ok, I understand why you changed that and added @fired_upon = false to the cell's attributes. One way to clean up the syntax a bit would be to write: 
```
@fired_upon = true 
@ship.hit if !empty?
```